From f6d7a5cafc1a9bb5514ed931ad90a3bbbae8095e Mon Sep 17 00:00:00 2001
From: Bingwu Zhang <xtex@aosc.io>
Date: Sat, 7 Jun 2025 23:23:20 +0800
Subject: [PATCH 120/133] link.Elf: allow multiple relocs at the same location
X-Developer-Signature: v=1; a=openpgp-sha256; l=877; i=xtex@astrafall.org;
 h=from:subject; bh=1ozaw381MW96JDe3PqYWEl5uq4PzNXZglIl9PnyACEY=;
 b=kA0DAAoWuRgIbtgEW5EByyZiAGm+bg6hSzUzFMIiDtWAdEqCAmJlbl0qq5Stb1ul2rwrYVvxm
 IiQBAAWCgA5FiEEcjGASwUsZw8VpncduRgIbtgEW5EFAmm+bg4bFIAAAAAABAAObWFudTIsMi41
 KzEuMTEsMiwyAAoJELkYCG7YBFuRomsA+M06sQiwXvW9fDmtwtVYqVuEuoh0z4akuWlCL3dP8n8
 BANR330m62co9ZbZzfXWr07Up5ndKq2EqL2rKTwW6N+YK
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

RISC-V and LoongArch ELF psABIs define a kind of
RELAX relocations which are expected to have a normal
relocation at the same address.

Change-Id: I5737bfcfd3e5041fb6ab7d193c9fc57eeca1eec8
(cherry picked from commit 8d06d8dca214eb9a07d1cc0f67374cdf35031db0)
---
 src/link/Elf/relocatable.zig | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/link/Elf/relocatable.zig b/src/link/Elf/relocatable.zig
index 58f18bd08bf8..11402ef58cd9 100644
--- a/src/link/Elf/relocatable.zig
+++ b/src/link/Elf/relocatable.zig
@@ -365,7 +365,6 @@ fn writeSyntheticSections(elf_file: *Elf) !void {
     const SortRelocs = struct {
         pub fn lessThan(ctx: void, lhs: elf.Elf64_Rela, rhs: elf.Elf64_Rela) bool {
             _ = ctx;
-            assert(lhs.r_offset != rhs.r_offset);
             return lhs.r_offset < rhs.r_offset;
         }
     };
-- 
2.52.0

